apps/checkout-app/backend/utils.ts

```diff
@@ -1,54 +1,54 @@
 import { PermissionEnum } from "@/checkout-app/graphql";
 import { NextApiRequest, NextApiResponse } from "next";
-import { debugEnvVars } from "../constants";
+import { debugEnvVars, envVars } from "../constants";
 import { isAuthenticated, isAuthorized } from "./auth";
 
 export const allowCors =
   (fn: (req: NextApiRequest, res: NextApiResponse) => Promise<void>) =>
-    async (req: NextApiRequest, res: NextApiResponse) => {
-      res.setHeader("Access-Control-Allow-Origin", req.headers.origin || "*");
-      res.setHeader("Access-Control-Allow-Methods", "GET,OPTIONS,POST");
-      res.setHeader(
-        "Access-Control-Allow-Headers",
-        "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version"
-      );
-
-      if (req.method === "OPTIONS") {
-        res.status(200).end();
-        return;
-      }
-
-      return await fn(req, res);
-    };
+  async (req: NextApiRequest, res: NextApiResponse) => {
+    res.setHeader("Access-Control-Allow-Origin", req.headers.origin || "*");
+    res.setHeader("Access-Control-Allow-Methods", "GET,OPTIONS,POST");
+    res.setHeader(
+      "Access-Control-Allow-Headers",
+      "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version"
+    );
+
+    if (req.method === "OPTIONS") {
+      res.status(200).end();
+      return;
+    }
+
+    return await fn(req, res);
+  };
 
 export const requireAuthorization =
```

Comment:
breaking SRP, maybe 2 middlewares, one for authentication, one for authorization, then compose them?

Reply:
For context: this is pretty old, before we had app-template 😅
New routes like this webhook handler should be written using retes, a library by zaiste. We already do that in app-template.

Honestly, imho this shouldn't even be in checkout-app and instead it should be in [saleor-app-sdk](https://github.com/saleor/saleor-app-sdk/blob/main/src/middleware.ts) or even [retes](https://github.com/kreteshq/retes/blob/main/src/middleware.ts).

@zaiste what do you think? I could write PR with CORS handler.

Retes has `withMethod` middleware which was used in this rewrite of webhook handler

Reply:
I wouldn't however change everything in this PR 😉 - I've opened a ticket in our backlog for rewriting the rest of API routes: SALEOR-7627

Reply:
Agree, just like mentioned, added few comments that are out of scope, so you can explain me context. So non blocking on my side